recipe/meta.yaml

```diff
@@ -76,3 +86,5 @@ extra:
     - shelnutt2
     - ihnorton
     - nguyenv
+  skip-lints:
+    - no_git_on_windows
```

Comment:
This should not be necessary

Reply:
git is used by vcpkg to download it's source.
I've decided to leave vcpkg enabled on windows (ie not unvendor it) as disabling it (and removing git) would require building libmagic (or file-feedstock) for windows.
They are doing it internally with some patches to libmagic source code that make is possible:
https://github.com/TileDB-Inc/TileDB/tree/2.26.2/ports/libmagic
Generally, I agree it's not the cleanest solution, but delivering it became more of a priority than cleanness.

Reply:
The selector changed. It was previously `not win` and now it's `win`... The lint should be respected. Please don't add `git` as a dependency on Windows. It'll cause hard to debug issues. The windows instances will have git already installed.

Reply:
In this case we would need to build libmagic for windows

Reply:
The instance already has git. Just remove it from the dependencies and you the build system should be able ot use the one installed on the system itself.

Reply:
Does the linter complain because it expects `git` in the `build` section but `git` is in an unusual `host` section here?

Reply:
The message states it's just `git` on windows that's an issue. There's another one (warning) that would like to see `git` in `build` and not `host`.
I'll try without `git`, but I believe it was failing in such case.

Reply:
An error on `win-64:`
```
CMake Error at build/_deps/vcpkg-src/scripts/buildsystems/vcpkg.cmake:899 (message):

  vcpkg install failed.  See logs for more information:

  C:\b\abs_86yj7i6d8o\croot\tiledb_1731056541602\work\build\vcpkg-bootstrap.log

Call Stack (most recent call first):

  C:/b/abs_86yj7i6d8o/croot/tiledb_1731056541602/_build_env/Library/share/cmake-3.26/Modules/CMakeDetermineSystem.cmake:148 (include)

  CMakeLists.txt:115 (project)

CMake Error: CMake was unable to find a build program corresponding to "Ninja".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.

CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage

CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
```

Reply:
Nope, I've tried this too. The message is misleading, the issue lays in `vcpkg install failed` part - that's due to CMake being unable to execute:
```
FetchContent_Declare(
            vcpkg
            GIT_REPOSITORY https://github.com/microsoft/vcpkg.git
            GIT_TAG        ${VCPKG_COMMIT_STRING}
            )
FetchContent_MakeAvailable(vcpkg)
```